lib/test/test_hamburger.rb

```diff
@@ -0,0 +1,272 @@
+    assert visibility[:show_student_options]    == Hamburger::HIDE_ALWAYS
+    assert visibility[:show_signed_out_options] == Hamburger::HIDE_ALWAYS
+    assert visibility[:show_pegasus_options]    == Hamburger::HIDE_ALWAYS
+    assert visibility[:show_help_options]       == Hamburger::SHOW_ALWAYS
+  end
+
+  def test_nonlevel_student_en
+    visibility = Hamburger.get_visibility({level: false, user_type: "student", language: "en"})
+
+    assert visibility[:hamburger_class]         == Hamburger::SHOW_ALWAYS
+    assert visibility[:show_teacher_options]    == Hamburger::HIDE_ALWAYS
+    assert visibility[:show_student_options]    == Hamburger::SHOW_MOBILE
+    assert visibility[:show_signed_out_options] == Hamburger::HIDE_ALWAYS
+    assert visibility[:show_pegasus_options]    == Hamburger::SHOW_ALWAYS
+    assert visibility[:show_help_options]       == Hamburger::SHOW_ALWAYS
+  end
+
+  def test_nonlevel_student_nonen
+    visibility = Hamburger.get_visibility({level: false, user_type: "student", language: "fr"})
+
+    assert visibility[:hamburger_class]         == Hamburger::SHOW_ALWAYS
+    assert visibility[:show_teacher_options]    == Hamburger::HIDE_ALWAYS
+    assert visibility[:show_student_options]    == Hamburger::SHOW_MOBILE
+    assert visibility[:show_signed_out_options] == Hamburger::HIDE_ALWAYS
+    assert visibility[:show_pegasus_options]    == Hamburger::HIDE_ALWAYS
+    assert visibility[:show_help_options]       == Hamburger::SHOW_ALWAYS
+  end
+
+  def test_nonlevel_nobody_en
+    visibility = Hamburger.get_visibility({level: false, user_type: nil, language: "en"})
+
+    assert visibility[:hamburger_class]         == Hamburger::SHOW_MOBILE
+    assert visibility[:show_teacher_options]    == Hamburger::HIDE_ALWAYS
+    assert visibility[:show_student_options]    == Hamburger::HIDE_ALWAYS
+    assert visibility[:show_signed_out_options] == Hamburger::SHOW_MOBILE
+    assert visibility[:show_pegasus_options]    == Hamburger::SHOW_MOBILE
+    assert visibility[:show_help_options]       == Hamburger::SHOW_MOBILE
+  end
+
+  def test_nonlevel_nobody_nonen
+    visibility = Hamburger.get_visibility({level: false, user_type: nil, language: "fr"})
+
+    assert visibility[:hamburger_class]         == Hamburger::SHOW_ALWAYS
+    assert visibility[:show_teacher_options]    == Hamburger::HIDE_ALWAYS
+    assert visibility[:show_student_options]    == Hamburger::HIDE_ALWAYS
+    assert visibility[:show_signed_out_options] == Hamburger::SHOW_MOBILE
+    assert visibility[:show_pegasus_options]    == Hamburger::HIDE_ALWAYS
+    assert visibility[:show_help_options]       == Hamburger::SHOW_ALWAYS
+  end
+
+  # Hamburger content tests.
+
+  def test_hamburger_content_teacher
+    contents = Hamburger.get_hamburger_contents({level: nil, script_level: nil, user_type: "teacher", language: "en"})
+    assert contents[:entries].find {|e| e[:id] == "hamburger-teacher-sections"}
```

Comment:
Does that work for just checking one field inside the hash (of which we have an array)?